kms/awskms/awskms_test.go

```diff
@@ -362,3 +338,54 @@ func Test_parseKeyID(t *testing.T) {
 		})
 	}
 }
+
+func Test_getCustomerMasterKeySpecMapping(t *testing.T) {
+	tmp := customerMasterKeySpecMapping
+	t.Cleanup(func() {
+		customerMasterKeySpecMapping = tmp
+	})
+
+	// Fail type switch
+	customerMasterKeySpecMapping[apiv1.SignatureAlgorithm(100)] = "string"
+
+	type args struct {
+		alg  apiv1.SignatureAlgorithm
+		bits int
+	}
+	tests := []struct {
+		name      string
+		args      args
+		want      types.KeySpec
+		assertion assert.ErrorAssertionFunc
+	}{
+		{"UnspecifiedSignAlgorithm", args{apiv1.UnspecifiedSignAlgorithm, 0}, types.KeySpecEccNistP256, assert.NoError},
+		{"SHA256WithRSA", args{apiv1.SHA256WithRSA, 0}, types.KeySpecRsa3072, assert.NoError},
+		{"SHA256WithRSA+2048", args{apiv1.SHA256WithRSA, 2048}, types.KeySpecRsa2048, assert.NoError},
+		{"SHA256WithRSA+3072", args{apiv1.SHA256WithRSA, 3072}, types.KeySpecRsa3072, assert.NoError},
+		{"SHA256WithRSA+4096", args{apiv1.SHA256WithRSA, 4096}, types.KeySpecRsa4096, assert.NoError},
+		{"SHA512WithRSA", args{apiv1.SHA512WithRSA, 0}, types.KeySpecRsa3072, assert.NoError},
+		{"SHA512WithRSA+2048", args{apiv1.SHA256WithRSA, 2048}, types.KeySpecRsa2048, assert.NoError},
+		{"SHA512WithRSA+3072", args{apiv1.SHA256WithRSA, 3072}, types.KeySpecRsa3072, assert.NoError},
+		{"SHA512WithRSA+4096", args{apiv1.SHA256WithRSA, 4096}, types.KeySpecRsa4096, assert.NoError},
```

Comment:
I think those should have `apiv1.SHA512WithRSA` for the names to match??

Reply:
Copy paste. They succeed because AWS is not like GCP, and you can use the same key for different types of signatures. Fixed with 3404685